src/structure/model.jl

```diff
@@ -222,378 +299,574 @@ end
-Base.getindex(struc::ProteinStructure, model_number::Integer) = struc.models[model_number]
-function Base.setindex!(struc::ProteinStructure, model::Model, model_number::Integer)
-    struc.models[model_number] = model
+Base.getindex(struc::ProteinStructure, mod_n::Integer) = struc.models[mod_n]
+
+function Base.setindex!(struc::ProteinStructure, mod::Model, mod_n::Integer)
+    struc.models[mod_n] = mod
     return struc
 end
 
 # Accessing a ProteinStructure with a Char returns the Chain with that chain ID
 #   on the default model
-Base.getindex(struc::ProteinStructure, chain_id::Char) = defaultmodel(struc)[chain_id]
-function Base.setindex!(struc::ProteinStructure, chain::Chain, chain_id::Char)
-    defaultmodel(struc)[chain_id] = chain
+Base.getindex(struc::ProteinStructure, ch_id::Char) = defaultmodel(struc)[ch_id]
+
+function Base.setindex!(struc::ProteinStructure, ch::Chain, ch_id::Char)
+    defaultmodel(struc)[ch_id] = ch
     return struc
 end
 
+# Check if an atom name exists in a residue as a whitespace-padded version
+function findatombyname(res::Residue, atom_name::AbstractString)
+    # Look for atom name directly
+    if atom_name in keys(res.atoms)
+        return res.atoms[atom_name]
+    # Pad out name to 4 characters to read PDB atom names with whitespace
+    elseif length(atom_name) == 3
+        if " $atom_name" in keys(res.atoms)
+            return res.atoms[" $atom_name"]
+        elseif "$atom_name " in keys(res.atoms)
+            return res.atoms["$atom_name "]
+        end
+    elseif length(atom_name) == 2
+        if " $atom_name " in keys(res.atoms)
+            return res.atoms[" $atom_name "]
+        elseif "  $atom_name" in keys(res.atoms)
+            return res.atoms["  $atom_name"]
+        elseif "$atom_name  " in keys(res.atoms)
+            return res.atoms["$atom_name  "]
+        end
+    elseif length(atom_name) == 1
+        if " $atom_name  " in keys(res.atoms)
+            return res.atoms[" $atom_name  "]
+        elseif "  $atom_name " in keys(res.atoms)
+            return res.atoms["  $atom_name "]
+        elseif "   $atom_name" in keys(res.atoms)
+            return res.atoms["   $atom_name"]
+        elseif "$atom_name   " in keys(res.atoms)
+            return res.atoms["$atom_name   "]
+        end
+    end
+    # Could not find atom name
+    throw(KeyError(atom_name))
+end
+
 
 # Getters and setters for structural elements
 
-# Atom getters/setters
+"Get the serial number of an `AbstractAtom`."
+serial(at::Atom) = at.serial
+serial(dis_at::DisorderedAtom) = serial(defaultatom(dis_at))
+
 
 """
-Determines if an `AbstractAtom` represents a non-hetero atom, i.e. came from an
-ATOM record.
+Get the atom name of an `AbstractAtom`.
+`spaces` determines whether surrounding whitespace is retained (default
+`false`).
 """
-ishetatom(atom::Atom) = atom.het_atom
+function atomname(at::Atom; spaces::Bool=false)
+    if spaces
+        return at.name
+    else
+        return strip(at.name)
```

Comment:
If you choose `strip` as the keyword argument name, you can call this as `Base.strip(at.name)`.